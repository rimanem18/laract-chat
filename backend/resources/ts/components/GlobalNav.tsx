import React from 'react'
import { Link } from 'react-router-dom'
import { useAppDispatch, useUserId } from '../app/hooks'
import { logout } from '../slices/AuthSlice'

const GlobalNav = () => {
  const userId = useUserId()
  const dispatch = useAppDispatch()

  // ログアウト
  const logoutHandler = () => {
    dispatch(logout())
  }

  console.log('GlobalNav')
  console.log(userId)

  return (
    <nav>
      <ul data-testid="nav">
        <Link to="/groups/1">
          <li>Top</li>
        </Link>
        <Link to="/about">
          <li>About</li>
        </Link>
        {userId === 0 ? (
          <>
            <Link to="/login">
              <li>Login</li>
            </Link>
            <Link to="/register">
              <li>Register</li>
            </Link>
          </>
        ) : (
          <a href="" onClick={logoutHandler}>
            <li>Logout</li>
          </a>
        )}
      </ul>
    </nav>
  )
}

export default React.memo(GlobalNav)
