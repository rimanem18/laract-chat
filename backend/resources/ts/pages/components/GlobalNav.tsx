import React from 'react'
import { Link } from 'react-router-dom'

const GlobalNav = () => {
  return (
    <nav>
      <ul>
        <Link to="/">
          <li>Top</li>
        </Link>
        <Link to="/about">
          <li>About</li>
        </Link>
        <Link to="/login">
          <li>Login</li>
        </Link>
        <Link to="/register">
          <li>Register</li>
        </Link>
      </ul>
    </nav>
  )
}

export default GlobalNav
