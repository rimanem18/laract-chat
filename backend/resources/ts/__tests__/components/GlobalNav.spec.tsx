import React from 'react'
import '@testing-library/jest-dom'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import { fireEvent, render } from '@testing-library/react'
import GlobalNav from '../../components/GlobalNav'

// Hooks の Mock
const mockUseAppDispatch = jest.fn()
const mockUseAppSelector = jest.fn()
jest.mock('../../app/hooks', () => ({
  useAppDispatch:
    () =>
    (...args: any[]) =>
      mockUseAppDispatch(...args),
  useAppSelector:
    () =>
    (...args: any[]) =>
      mockUseAppSelector(...args),
  useUserId: () => useUserIdMock(),
}))

// Mock の定義
const useUserIdMock = jest.fn().mockReturnValue(1)

// Setup
const setup = () => {
  const screen = render(
    <Router>
      <GlobalNav />
    </Router>
  )
  const nav = screen.getByTestId('nav') as HTMLElement
  return {
    nav,
    ...screen,
  }
}

// テスト開始
describe('GlobalNav', () => {
  it('ログインしているときだけログアウトボタンがある', () => {
    const { nav } = setup()

    expect(Boolean(nav.textContent?.match(/Logout/g))).toBe(true)
  })

  it('ログインしていると登録ボタンとログインボタンは表示されない', () => {
    const { nav } = setup()

    expect(Boolean(nav.textContent?.match(/Login/g))).toBe(false)
    expect(Boolean(nav.textContent?.match(/Register/g))).toBe(false)
  })
})
