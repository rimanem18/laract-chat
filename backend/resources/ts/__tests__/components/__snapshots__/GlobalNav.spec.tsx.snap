// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalNav snapshot 1`] = `
<nav>
  <ul
    data-testid="nav"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <li>
        Top
      </li>
    </a>
    <a
      href="/about"
      onClick={[Function]}
    >
      <li>
        About
      </li>
    </a>
    <a
      href=""
      onClick={[Function]}
    >
      <li>
        Logout
      </li>
    </a>
  </ul>
</nav>
`;
